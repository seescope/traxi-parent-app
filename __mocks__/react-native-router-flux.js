import React, { PropTypes } from 'react';
import { View } from 'react-native';

export const Actions = {
  createKid: jest.fn(),
  walkthrough: jest.fn(),
  congratulations: jest.fn(),
  thankyou: jest.fn(),
  areYouReady: jest.fn(),
  intro: jest.fn(),
};

export const Scene = (props) => <View {...props} />;

export const Router = ({ children }) => (
  <View>{children}</View>
);

Router.propTypes = {
  children: PropTypes.array.isRequired,
};
