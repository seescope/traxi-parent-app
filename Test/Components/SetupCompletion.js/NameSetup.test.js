import { Alert } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Actions } from 'react-native-router-flux';

import {
  NameSetup,
  nextStep,
} from '../../../App/Components/SetupCompletion/NameSetup';

Alert.alert = jest.fn();
Actions.kidImageSetup = jest.fn();

describe('NameSetup', () => {
  it('Renders correctly', () => {
    const setName = jest.fn();

    const tree = renderer.create(
      <NameSetup name={'Chris'} setNameFn={setName} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('nextStep() alerts when no name was entered', () => {
    nextStep();
    expect(Alert.alert).toHaveBeenCalledWith('Please enter your name');
  });

  it('nextStep() goes to kidImageSetup when there is a name ', () => {
    nextStep('Chris');
    expect(Actions.kidImageSetup).toHaveBeenCalled();
  });
});
