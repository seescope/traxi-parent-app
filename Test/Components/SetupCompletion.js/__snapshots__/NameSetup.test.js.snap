// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameSetup Renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#448AFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 6,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFF",
          "fontFamily": "Raleway-ExtraBold",
          "fontSize": 30,
          "marginBottom": -2,
          "textAlign": "center",
        }
      }
    >
      Welcome !
    </Text>
    <View
      style={
        Object {
          "height": 32,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontFamily": "Raleway-Regular",
          "fontSize": 16,
        }
      }
    >
      What is your name
    </Text>
    <View
      style={
        Object {
          "height": 10,
        }
      }
    />
    <TextInput
      autoCapitalize="words"
      autoCorrect={false}
      keyboardType={undefined}
      onChangeText={[Function]}
      onSubmitEditing={undefined}
      placeholderTextColor="rgba(75, 73, 73, 0.8)"
      secureTextEntry={undefined}
      style={
        Object {
          "backgroundColor": "#EFEFEF",
          "borderRadius": 4,
          "color": "rgba(75, 73, 73, 1)",
          "fontFamily": "Raleway-Regular",
          "height": 32,
          "paddingHorizontal": 8,
          "width": 686,
        }
      }
      underlineColorAndroid="rgba(255, 255, 255, 1)"
      value={undefined}
    />
    <View
      style={
        Object {
          "height": 24,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 20,
          }
        }
      >
        Next Step
         
      </Text>
    </View>
  </View>
</View>
`;

exports[`NameSetup nextStep() alerts when no name was entered 1`] = `undefined`;

exports[`Renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#448AFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 6,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFF",
          "fontFamily": "Raleway-ExtraBold",
          "fontSize": 30,
          "marginBottom": -2,
          "textAlign": "center",
        }
      }
    >
      Welcome !
    </Text>
    <View
      style={
        Object {
          "height": 32,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontFamily": "Raleway-Regular",
          "fontSize": 16,
        }
      }
    >
      What is your name
    </Text>
    <View
      style={
        Object {
          "height": 10,
        }
      }
    />
    <TextInput
      autoCapitalize="words"
      autoCorrect={false}
      keyboardType={undefined}
      onChangeText={[Function]}
      onSubmitEditing={undefined}
      placeholderTextColor="rgba(75, 73, 73, 0.8)"
      secureTextEntry={undefined}
      style={
        Object {
          "backgroundColor": "#EFEFEF",
          "borderRadius": 4,
          "color": "rgba(75, 73, 73, 1)",
          "fontFamily": "Raleway-Regular",
          "height": 32,
          "paddingHorizontal": 8,
          "width": 686,
        }
      }
      underlineColorAndroid="rgba(255, 255, 255, 1)"
      value={undefined}
    />
    <View
      style={
        Object {
          "height": 24,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 20,
          }
        }
      >
        Next Step
         
      </Text>
    </View>
  </View>
</View>
`;
